core/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -260,6 +260,7 @@ private void processBatch(Segment segment, MessageStream<TrackedEventMessage<?>>
                 if (batch.isEmpty()) {
                     TrackingToken finalLastToken = lastToken;
                     transactionManager.executeInTransaction(() -> tokenStore.storeToken(finalLastToken, getName(), segment.getSegmentId()));
+                    activeSegments.computeIfPresent(segment.getSegmentId(), (k, v) -> v.advancedTo(finalLastToken));
```

Comment:
So if the `eventStream` for a given segment does not have any new events to process, this segment will not get advanced to `finalLastToken`? Is my understanding correct?

Reply:
`finalLastToken` is always updated and represents the last token seen in any event, whether it was handled, or not. The problem is/was, that this last token was only updated in the local "activeSegments" map (which represents the processing status for this processor) when a batch was executed. So if a full batch of messages was unhandled, the token seemed to stand still, while in reality (and in the tokenstore), is _was_ being updated.